### PR TITLE
expidus.file-manager: init at 0.1.2

### PR DESCRIPTION
**Pull request by RossComputerGuy**
###### Description of changes

Adds ExpidusOS's File Manager.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes (or backporting 23.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/desktops/expidus/file-manager/default.nix [hunk 1]
```diff
@@ -0,0 +1,49 @@
+{ lib, flutter, fetchFromGitHub }:
+with lib;
+let
+  version = "0.1.2";
+in flutter.buildFlutterApplication {
+  pname = "expidus-file-manager";
+  inherit version;
```
Comment (SuperSandro2000) [inline pkgs/desktops/expidus/file-manager/default.nix:7]:
```suggestion
{ lib, flutter, fetchFromGitHub }:

flutter.buildFlutterApplication rec {
  pname = "expidus-file-manager";
  version = "0.1.2";
```

File: pkgs/desktops/expidus/file-manager/default.nix [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    rev = version;
+    sha256 = "sha256-aAPmwzNPgu08Ov9NyRW5bcj3jQzG9rpWwrABRyK2Weg=";
+  };
+
+  depsListFile = ./deps.json;
+  vendorHash = "sha256-mPGrpMUguM9XAYWH8lBQuytxZ3J0gS2XOMPkKyFMLbc=";
+
+  postInstall = ''
+    rm $out/bin/file_manager
+    ln -s $out/app/file_manager $out/bin/expidus-file-manager
```
Comment (SuperSandro2000) [inline pkgs/desktops/expidus/file-manager/default.nix:21]:
```suggestion
    ln -fs $out/app/file_manager $out/bin/expidus-file-manager
```

File: pkgs/desktops/expidus/default.nix [hunk 3]
```diff
@@ -0,0 +1,5 @@
+{ lib, callPackage }:
+with lib;
+genAttrs [
+  "file-manager"
+] (name: callPackage (./. + "/${name}") {})
```
[the remark this answers is not in the document]
Comment (RossComputerGuy) [reply pkgs/desktops/expidus/default.nix]: Done

File: pkgs/desktops/expidus/file-manager/default.nix (hunk 1 above)
Comment (RossComputerGuy) [reply pkgs/desktops/expidus/file-manager/default.nix]: Done

File: pkgs/desktops/expidus/file-manager/default.nix [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+
+    mkdir -p $out/share/metainfo
+    mv $out/app/data/com.expidusos.file_manager.metainfo.xml $out/share/metainfo
+
+    substituteInPlace "$out/share/applications/com.expidusos.file_manager.desktop" \
+      --replace "Exec=file_manager" "Exec=$out/bin/expidus-file-manager" \
+      --replace "Icon=com.expidusos.file_manager" "Icon=$out/share/icons/com.expidusos.file_manager.png"
+  '';
+
+  disallowedReferences = with pkgs; [
```
[the remark this answers is not in the document]
Comment (RossComputerGuy) [reply pkgs/desktops/expidus/file-manager/default.nix]: Probably should be, done.

File: pkgs/desktops/expidus/file-manager/default.nix [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+    flutter
+    flutter.unwrapped
+  ];
+
+  meta = {
```
[the remark this answers is not in the document]
Comment (RossComputerGuy) [reply pkgs/desktops/expidus/file-manager/default.nix]: Done

File: pkgs/desktops/expidus/file-manager/default.nix (hunk 2 above)
Comment (RossComputerGuy) [reply pkgs/desktops/expidus/file-manager/default.nix]: Actually, that keeps 2 executables which that shouldn't be done.
```
$ ls result/bin
expidus-file-manager  file_manager
```

File: pkgs/top-level/all-packages.nix [hunk 6]
```diff
@@ -37554,6 +37554,8 @@ with pkgs;
 
   enlightenment = recurseIntoAttrs (callPackage ../desktops/enlightenment { });
 
+  expidus = callPackage ../desktops/expidus {};
```
Comment (SuperSandro2000) [inline pkgs/top-level/all-packages.nix:37557]: I think we need a recurseIntoAttrs or similar here otherwise ofborg is not building the package

Comment (RossComputerGuy) [reply pkgs/top-level/all-packages.nix:37557]: Done

**Comment by SuperSandro2000** [comment]
@ofborg build expidus.file-manager

File: pkgs/desktops/expidus/file-manager/default.nix [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+    mv $out/app/data/com.expidusos.file_manager.metainfo.xml $out/share/metainfo
+
+    substituteInPlace "$out/share/applications/com.expidusos.file_manager.desktop" \
+      --replace "Exec=file_manager" "Exec=$out/bin/expidus-file-manager" \
+      --replace "Icon=com.expidusos.file_manager" "Icon=$out/share/icons/com.expidusos.file_manager.png"
+  '';
+
+  disallowedReferences = [
+    flutter
+    flutter.unwrapped
```
Comment (SuperSandro2000) [inline pkgs/desktops/expidus/file-manager/default.nix:36]:
```
error: output '/nix/store/d98grmgli8psici4ll1j5zy96rrr80rp-expidus-file-manager-0.1.2-debug' is not allowed to refer to the following paths:
         /nix/store/jkx6ckjhybkmiq80nlbakv5hln38s4dn-flutter-3.10.5-unwrapped
```

Comment (RossComputerGuy) [reply pkgs/desktops/expidus/file-manager/default.nix:43]: Huh, didn't get that locally. I pushed a fix.

**Comment by RossComputerGuy** [comment]
Huh, it still says the reference isn't allowed but I don't see this on my local machine.
```
$ nix eval '.#legacyPackages.x86_64-linux.flutter.unwrapped.outPath'
"/nix/store/6pzqx3mzr7yirgdqsn2v907z15pd30g7-flutter-3.10.0-unwrapped"

$ grep -r "/nix/store/6pzqx3mzr7yirgdqsn2v907z15pd30g7-flutter-3.10.0-unwrapped" result-debug/app.linux-x64.symbols
FAIL: 1

$ grep -r "/nix/store/6pzqx3mzr7yirgdqsn2v907z15pd30g7-flutter-3.10.0-unwrapped" result/
FAIL: 1
```

**Pull request merged**